Classes/common/touchdb/TDMisc.m

```diff
@@ -209,12 +209,15 @@ BOOL TDMayBeTransientError(NSError* error)
 {
     NSString* domain = error.domain;
     NSInteger code = error.code;
-    if ($equal(domain, NSURLErrorDomain)) {
+    if ([domain isEqualToString:NSURLErrorDomain]) {
         return code == NSURLErrorTimedOut || code == NSURLErrorCannotConnectToHost ||
                code == NSURLErrorNetworkConnectionLost;
-    } else if ($equal(domain, TDHTTPErrorDomain)) {
+    } else if ([domain isEqualToString:TDHTTPErrorDomain]) {
         // Internal Server Error, Bad Gateway, Service Unavailable or Gateway Timeout:
         return code == 500 || code == 502 || code == 503 || code == 504;
+    } else if ([domain isEqualToString:NSPOSIXErrorDomain]) {
+        //connection reset by peer
+        return code == 54;
```

Comment:
magic number here with no explanation - i think we looked at this and found it in `errno.h` ?